### PR TITLE
Feature Detection Speedups

### PR DESCRIPTION
**Pull request by freemansw1**
Similar to #89, these are our speedups to feature detection. These changes are overall a bit more complicated and the result less dramatic than #89, but these changes still provide about a 100x speedup vs. the current `main` and `dev` branches and a ~2x speedup vs `v2.0-dev`. This again shows a benchmark, using the same machine and dataset as #89: 
feature_time_nolog

### Change Summary

`feature_position`
Switched this code to use a subset of the input track data rather than performing operations on the array every time. This is particularly important for large arrays (and will be more important when 3D feature detection is added)

`remove_parents` 
Entirely new function that operates on 1D coordinates rather than 2D tuples, as that is substantially less computationally expensive. Required changes to `feature_detection_threshold`, as well, to return 1D points.  

`feature_detection_threshold`
Extensive use of the `skimage.measure.regionprops` function to get information about the various labeled regions, which is substantially faster than the previous approach. New utility functions added to `tobac.utils` to help this process.

### CODE REVIEW

File: tobac/utils.py [hunk 1]
```diff
@@ -509,3 +509,96 @@ def get_spacings(field_in, grid_spacing=None, time_spacing=None):
...
+    for region_prop_key in region_property_dict:
+        region_prop = region_property_dict[region_prop_key]
+        index = region_prop.label
+        curr_y_ixs, curr_x_ixs = np.transpose(region_prop.coords)
+
+        y_indices[index] = curr_y_ixs
+        x_indices[index] = curr_x_ixs
+        curr_loc_indices[index] = len(curr_y_ixs)
+
+    return [curr_loc_indices, y_indices, x_indices]
```
Comment (w-k-jones) [inline tobac/utils.py:580]: Would returning as a tuple be better here?

File: tobac/feature_detection.py [hunk 2]
```diff
@@ -1,41 +1,70 @@
 import logging
 import numpy as np
 import pandas as pd
+from . import utils as tb_utils
 
 
 def feature_position(
     hdim1_indices,
     hdim2_indeces,
```
[the remark this answers is not in the document]
Comment (freemansw1) [reply tobac/feature_detection.py]: I've fixed this in my new commit

File: tobac/utils.py [hunk 3]
```diff
@@ -509,3 +509,96 @@ def get_spacings(field_in, grid_spacing=None, time_spacing=None):
...
+    -------
+    dict
+        output from skimage.measure.regionprops in dictionary format,
+        where they key is the label number
+    """
+    import skimage.measure
+
+    region_properties_raw = skimage.measure.regionprops(labels)
+    region_properties_dict = dict()
+    for region_prop in region_properties_raw:
```
[the remark this answers is not in the document]
Comment (freemansw1) [reply tobac/utils.py]: I ended up deleting this as it was a duplicated function

File: tobac/utils.py (hunk 1 above)
Comment (freemansw1) [reply tobac/utils.py]: I've switched this as we discussed.

File: tobac/utils.py [hunk 4]
```diff
@@ -509,3 +509,96 @@ def get_spacings(field_in, grid_spacing=None, time_spacing=None):
...
+    Returns
+    -------
+    dict
+        output from skimage.measure.regionprops in dictionary format, where they key is the label number
+    """
+    import skimage.measure
+
+    region_properties_raw = skimage.measure.regionprops(labels)
+    region_properties_dict = dict()
+    for region_prop in region_properties_raw:
```
[the remark this answers is not in the document]
Comment (freemansw1) [reply tobac/utils.py]: Yes, agreed! I've switched this style now.

**Review by w-k-jones — APPROVED** [review]

**Pull request merged**